### DISCUSSION TITLE
Feature request to rename and edit virtual audio devices

### DISCUSSION THREAD

**Discussion by cxy930123** [ExistentialAudio/BlackHole#608] [Q&A]
Do you have any plan to let user rename, add or delete virtual devices?

**Comment by devinroth** [comment]
Please read the wiki on renaming and running multiple devices.

Devin

**Comment by cxy930123** [comment]
> Please read the wiki on renaming and running multiple devices. Devin
> […](#)
> On Apr 19, 2022, 9:28 PM -0700, 陈兴宇 ***@***.***>, wrote: Do you have any plan to let user rename, add or delete virtual devices? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

I cannot find anything about rename an audio virtual device in wiki, but I find a way to add multiple devices by [Running Multiple BlackHole Drivers](https://github.com/ExistentialAudio/BlackHole/wiki/Running-Multiple-BlackHole-Drivers). However, I hope I can edit devices by commands in the terminal, not by building another driver in Xcode.

**Comment by devinroth** [comment]
You’re right. It’s not there.

Take a look at the source. It’s kDevice_Name.

Devin
